crates/evm/core/src/utils.rs

```diff
@@ -184,17 +184,29 @@ pub fn create2_handler_register<DB: revm::Database, I: InspectorExt<DB>>(
 
                 // Decode address from output.
                 let address = match outcome.instruction_result() {
-                    return_ok!() => Address::try_from(outcome.output().as_ref())
-                        .map_err(|_| {
+                    return_ok!() => {
+                        let output = outcome.output();
+
+                        // Here we need to handle both evm and zksync return data to parse the
+                        // address
+                        if output.len() == 20 {
+                            Some(Address::from_slice(output))
+                        } else if output.len() >= 32 {
```

Comment:
perhaps we can instead check on the factory that we are using? so instead of checking the return value we check if we are in "zksync" or not, and act accordingly.
the output is also usable as check, but I'd like to keep the link the zksync, instead of the output itself

Reply:
Please mark this as explicit with checking for zksync context.